src/commands/up.rs

```diff
@@ -308,29 +292,54 @@ impl UpCommand {
         Ok(locked_url)
     }
 
-    async fn prepare_app_from_file(
+    // Take the AppSource and do the minimum amount of work necessary to
+    // be able to resolve the trigger executor.
+    async fn resolve_app_source(
```

Comment:
(a note for the future, not for now)  This function and `load_resolved_app_source` now have almost nothing to do with `self` - it might be interesting to explore making AppSource and ResolvedAppSource more self-contained at some future point.  Not a consideration for now though

Reply:
Indeed `up.rs` seems ripe for major refactoring "pretty soon".